src/ocm/components/clusterConfiguration/staticIp/data/formDataToInfraEnvField.ts

```diff
@@ -74,36 +76,44 @@ const getNmstateObject = (
         ),
       };
       nicName = getDummyNicName(protocolVersion);
-      interfaces.push(getEthernetInterface(nicName, protocolConfigs));
+
+      if (networkWide.useVlan && networkWide.vlanId) {
+        const vlanInterface = getVlanInterface(nicName, networkWide.vlanId, protocolConfigs);
+        interfaces.push(vlanInterface);
+        nicName = vlanInterface.name;
+      } else {
+        interfaces.push(getEthernetInterface(nicName, protocolConfigs));
+      }
     }
+
     routeConfigs.push(
       getRouteConfig(protocolVersion, networkWide.ipConfigs[protocolVersion].gateway, nicName),
     );
   }
+
   if (Object.keys(realProtocolConfigs).length > 0) {
-    interfaces.unshift(getEthernetInterface(REAL_NIC_NAME, realProtocolConfigs));
-  }
-  if (networkWide.useVlan && networkWide.vlanId) {
-    let configs = [] as NmstateProtocolConfigs;
-    interfaces.forEach((item) => {
-      configs = {
-        ipv4: item['ipv4'] ? item['ipv4'] : configs['ipv4'],
-        ipv6: item['ipv6'] ? item['ipv6'] : configs['ipv6'],
-      };
-      delete item['ipv4'];
-      delete item['ipv6'];
-    });
-    const vlanInterface = getVlanInterface(interfaces[0].name, networkWide.vlanId, configs);
-    interfaces.push(vlanInterface);
-    routeConfigs.forEach((route) => {
-      route['next-hop-interface'] = vlanInterface.name;
-    });
+    if (networkWide.useVlan && networkWide.vlanId) {
+      const vlanInterface = getVlanInterface(
+        REAL_NIC_NAME,
+        networkWide.vlanId,
+        realProtocolConfigs,
+      );
+
+      interfaces.unshift(vlanInterface);
+      routeConfigs.forEach((route) => {
```

Comment:
updating routeConfigs for vlan is not required anymore, what's done in line 90 is already correct